components/support/error/macros/src/argument.rs

```diff
@@ -2,16 +2,33 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-pub(crate) fn validate(arguments: &syn::AttributeArgs) -> syn::Result<()> {
-    // For now we validate that there are no arguments.
-    // if we want to improve the macro to support attributes, for example,
-    // if we'd like to pass in the return type of the body we can modify this
-    // logic to parse and return the type.
-    if !arguments.is_empty() {
-        return Err(syn::Error::new(
+/// Returns the name of the type of the "internal" error.
+pub(crate) fn validate(arguments: &syn::AttributeArgs) -> syn::Result<String> {
+    // We insist on a single arg of form `E=ErrorType`. If we end up growing
+    // more args we'd probably just add a struct and return that, but this seems
+    // ok for now.
+    fn oops(inner: &str) -> syn::Error {
```

Comment:
a bit of a nit: We can have this function take a [`Span`](https://docs.rs/proc-macro2/latest/proc_macro2/struct.Span.html) and pass it to the error creation, which will make the errors a bit more localized!

(if we import the trait `Spanned` from `syn` we can get a `Span` from all the `syn` types!

ref: https://docs.rs/syn/latest/syn/spanned/trait.Spanned.html